x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/create_step/progress_stats.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiProgress, EuiSpacer, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useMlKibana } from '../../../../../contexts/kibana';
+import { getDataFrameAnalyticsProgressPhase } from '../../../analytics_management/components/analytics_list/common';
+import { isGetDataFrameAnalyticsStatsResponseOk } from '../../../analytics_management/services/analytics_service/get_analytics';
+import { ml } from '../../../../../services/ml_api_service';
+import { DataFrameAnalyticsId } from '../../../../common/analytics';
+
+export const PROGRESS_REFRESH_INTERVAL_MS = 1000;
+
+export const ProgressStats: FC<{ jobId: DataFrameAnalyticsId }> = ({ jobId }) => {
+  const [initialized, setInitialized] = useState<boolean>(false);
+  const [currentProgress, setCurrentProgress] = useState<
+    | {
+        currentPhase: number;
+        progress: number;
+        totalPhases: number;
+      }
+    | undefined
+  >(undefined);
+
+  const {
+    services: { notifications },
+  } = useMlKibana();
+
+  useEffect(() => {
+    setInitialized(true);
+  }, []);
+
+  useEffect(() => {
+    const interval = setInterval(async () => {
+      try {
+        const analyticsStats = await ml.dataFrameAnalytics.getDataFrameAnalyticsStats(jobId);
+        const jobStats = isGetDataFrameAnalyticsStatsResponseOk(analyticsStats)
+          ? analyticsStats.data_frame_analytics[0]
+          : undefined;
+
+        if (jobStats !== undefined) {
```

Comment:
do you need to clear the interval if `jobsStats` is `undefined`?

Reply:
Yes, updated in b799788f8112fb423b4154a8e4a3d7e361908a3b 👍